optaplanner-persistence/optaplanner-persistence-jaxb/src/test/java/org/optaplanner/persistence/jaxb/api/score/json/buildin/bendablebigdecimal/BendableBigDecimalScoreJaxbJsonAdapterTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.persistence.jaxb.api.score.json.buildin.bendablebigdecimal;
+
+import java.math.BigDecimal;
+
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+
+import org.junit.Test;
+import org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScore;
+import org.optaplanner.persistence.jaxb.api.score.buildin.bendablebigdecimal.BendableBigDecimalScoreJaxbXmlAdapter;
+import org.optaplanner.persistence.jaxb.api.score.json.AbstractScoreJaxbJsonAdapterTest;
+
+public class BendableBigDecimalScoreJaxbJsonAdapterTest extends AbstractScoreJaxbJsonAdapterTest {
```

Comment:
The other test, `SimpleScoreXMLJsonAdapterTest` is called like that because it is `SimpleScoreXMLJsonAdapter` + `Test`.

This json stuff, is also a test for the `SimpleScoreXMLJsonAdapter`, so it too, can go into the `SimpleScoreXMLJsonAdapterTest`... (yes, sounds weird). And it can reuse the wrapper.

Reply:
Doing that, will heavily reduce the number of code lines, making the code easier to maintain.

"Good programmers add code to add features. Great programmers remove code (without removing features)." (there's a better version of this quoth somewhere).

I wouldn't even mind that the we just adjust the superclass's serializeAndDeserialize() to call serializeAndDeserializeXML() and serializeAndDeserializeJson(), avoiding any changes to specific score tests

Reply:
Point noted; good quote! My thought process about the `SimpleScoreJaxbXMLAdapterTest` was that it tests the `SimpleScoreJaxbXMLAdapter`, so the test is specific to JAXB standard XML. I would argue that the JSON tests I added is a test for the `SimpleScoreJaxbJsonAdapter`, so the `SimpleScoreJaxbJsonAdapterTest` is responsible for JAXB standard JSON (not XML). That's why I put them in separate classes.

But I absolutely agree with you that there's a lot of code duplication here. The problem is that the `Marshaller` created for XML serialization is different from the `Marshaller` created for JSON serialization (because of the MEDIATYPE property that JSON needs). I like your `serializeAndDeserializeXML()` and `serializeAndDeserializeJson()` separation idea, I should be able to use this to avoid duplicating all the test classes.